system/include/emscripten/wasmfs.h

```diff
@@ -32,6 +35,10 @@ int wasmfs_create_file(const char* pathname __attribute__((nonnull)), mode_t mod
 // Returns 0 on success like `mkdir`, or a negative value on error.
 int wasmfs_create_directory(const char* path __attribute__((nonnull)), mode_t mode, backend_t backend);
 
+// Unmounts the directory at a specific path.
+// Returns 0 on success, or a negative value on error.
+int wasmfs_unmount(int dirfd, intptr_t path);
```

Comment:
It is slightly odd to have unmount but no mount, so at least a comment about that might make sense.

I wonder, though, if we have `create_directory` instead of `mount`, perhaps we could have `remove_directory` instead of unmount..? (With comments that they both do something like a mount/unmount if the backend differs from the parent dir perhaps.) Thoughts?

Reply:
Is it possible that adding `remove_directory` would add more confusion since we already have `__syscall_rmdir` and `__syscall_unlinkat`? I think the main reason we chose to add `wasmfs_unmount` here was because we were essentially removing the directory while handling a few special cases (Verifying that the child is actually a mountpoint, allowing unmounts/unlinks of nonempty directories, and not doing any path parsing).

Reply:
In hindsight, I wish we had named `wasmfs_create_directory` `wasmfs_mount` instead (and relatedly we should get rid of `wasmfs_create_file` and only allow mounting directories.)

Reply:
Is this something that would be difficult to change? If so, would it be worth the effort to manually go through and change every usage?

Reply:
The problem is that this is the public API for WasmFS, so simply renaming the existing function would break users. We can perhaps create an alias with the new name and mark the old version as deprecated and remove it after a transition period, but we should consider doing that as a separate PR.

Reply:
If we think we might add an alias `wasmfs_mount` later than leaving this as "unmount" for now sgtm. Maybe add a comment and a TODO though.